components/event-bus/internal/knative/util/kn-eventing.go

```diff
@@ -301,10 +301,7 @@ func makeHttpRequest(channel *evapisv1alpha1.Channel, headers *map[string]string
 		return nil, err
 	}
 	req.Header.Set("Content-Type", "application/json")
-
-	for k, v := range *headers {
-		req.Header.Set(k, v)
-	}
+	req.Header = *headers
```

Comment:
Please move this line before `req.Header.Set("Content-Type", "application/json")`, to not lose setting the content-type header.

Reply:
Thanks for the catch!
Fixed